### PR TITLE
dk/create user profile

### PR DESCRIPTION
**Pull request by kos1kov**

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,10 @@
 Rails.application.routes.draw do
   root 'welcome#index'
-  mount_devise_token_auth_for 'User', at: 'auth'
+  mount_devise_token_auth_for 'User', at: 'auth',controllers:  {
+
```
Comment (LuginaJulia) [inline config/routes.rb]: extra line

File: app/javascript/components/UserProfile.jsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      user: {
+        first_name: user.first_name,
+        last_name: user.last_name,
+        phone: user.phone,
+        photo: null
+      }
+    };
+    this.handleSave = this.handleSave.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
```
Comment (MargaretOm) [inline app/javascript/components/UserProfile.jsx]: Where handleChangeFile?

File: app/controllers/users/registrations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,215 @@
+require 'aws-sdk-s3'
```
Comment (svintbel) [inline app/controllers/users/registrations_controller.rb]: I think it should work without requiring.

File: config/credentials.yml.enc [hunk 4]
```diff
@@ -1 +1 @@
-u4vSs9rRKgFB9RgWv9p02XEvtP7XQ8zGXKUA4t7UyHCDIdHcoHmpSdjGSTUMScwW8jkTH6lUCVM9ZivZO0ds4o137PIRrZlIOTAVBRNRndgI3ZssOOhG5bkQhEzV5Aa3PhpNxrR4skUiE1VaTDDgZrCipez/63NAuO00G3p/KO2UrxyCyGT5IsNnyb3fPDAD9OiaVF/g1us4rrARgvwaUGB++8+QgPxFSMSdwt18WojUSj+j8b5+XgX36XxngdgnCF9Na3oSnH9HYvM+FzrjyubTUCj2t66Ll3vme4yIEufxWEkcFiv/2f4raFhd4fyEkR3f6tFP7F86UmBAbW5w3Ub+Mvt8OgLpy+TkvchOQGHOs7Myo2MFfE6zscFbOJU5GzgJl8v6faXiz1fn11DVOnNeidKN8RJEAUcD--gw8snR+zohnRIwCV--Orm93ffGg+Aq83DtSpdCAQ==
\ No newline at end of file
+j7MeyJH1xea0QCkb4TtzgYXytdhJHx17Nc5uUGTohXtaJ1e61K948piG4V2ZWQYapo2NK/3SBmXUP/DshSPdDPst8mcGthUwXYqBAPKOXapNanfsjAiu9s2rbKkqGAYiOuT8LWMk+LSqGdI0HAp49viXsicWjeBvTlCOwpiuCipZyfHOOck9N0/BFsU385uhigJ7Zq+F81HwtIWaAIM5b/TEi6uvPBwXw8AmesagLguWGJ/L5HIG5TSw9evo/r76zLlgzskFzCWATePKjy6w/ok7jFAyttrwUgUHgcKz4BqMopTYycul98vrtPp1vcVVUGUqrwCvA4dD+PYTbWXnNjZov4sHWVntUXmoNhbfBSdEfOYQtKeVB1ysWwxVfy7AfoGrx5QLVlD7bGEA9EAfB3BnfljEOLx45zYI4wVJjA4o20QZo7xoV+LidNs5xdkOvowLnEIeaz6RWjGfnGrzRsOFFVHH7mYz--Q46QUOmQziEbkcnf--2OgwWbr3RA6N/AkOO4EKnw==
```
Comment (svintbel) [inline config/credentials.yml.enc]: Is it ok if anyone can see this token?

Comment (kos1kov) [reply config/credentials.yml.enc]: credentials.yml.enc is encrypted with the secret key that's stored in master.key

File: app/controllers/users/registrations_controller.rb (hunk 3 above)
Comment (kos1kov) [reply app/controllers/users/registrations_controller.rb]: i test without and get error

File: spec/requests/edit_profile_spec.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+describe 'Edit profile', type: :request do
+  let(:user) { create(:user) }
+  let(:user_attributes) {
+    [
+      first_name: 'dima', last_name: 'kosikov', phone: '123456789',
```
Comment (anatoliliotych) [inline spec/requests/edit_profile_spec.rb]: how about faker?

File: app/controllers/users/registrations_controller.rb (hunk 3 above)
Comment (anatoliliotych) [inline app/controllers/users/registrations_controller.rb]: still think that something is wrong, probably the order of loading gems.

File: app/javascript/components/UserProfile.jsx [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    if (this.state.errors) {
+      errorMessages = <div>
+        {this.state.errors.map((error) => (
+          <p key={error.id}>{error}</p>
+        ))}
+      </div>;
+    }
+
+    let image;
+    let user_role;
```
Comment (ermaac) [inline app/javascript/components/UserProfile.jsx]: camelCase

File: config/credentials.yml.enc (hunk 4 above)
Comment (ermaac) [inline config/credentials.yml.enc]: EOF

File: spec/requests/edit_profile_spec.rb [hunk 7]
```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+describe 'Edit profile', type: :request do
+  let(:user) { create(:user) }
+  let(:user_attributes) {
+    [
```
Comment (ermaac) [inline spec/requests/edit_profile_spec.rb]: why it is array?

File: config/credentials.yml.enc (hunk 4 above)
Comment (kos1kov) [reply config/credentials.yml.enc]: not work if new line
image

**Pull request merged**